### DISCUSSION TITLE
ffs : AttributeError: type object 'Color' has no attribute 'parse'

### DISCUSSION THREAD

**Discussion by Kid-C** [smacke/subsync#170] [Q&A] [answered]
Hi. Thanks for this project, which seems very good.

I installed ffsubsync under Win10, but I get an error when executing. It's probably some dependency or something out-dated.

The program trace is included below. Any tips are appreciated. Best....

`
C:\0>python --version
Python 3.6.0

C:\0>git --version
git version 2.39.0.windows.1

C:\0>ffmpeg -version
ffmpeg version N-91116-g8e7b13b971 Copyright (c) 2000-2018 the FFmpeg developers
built with gcc 7.3.0 (GCC)
configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-bzlib --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-amf --enable-ffnvcodec --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth
libavutil      56. 18.102 / 56. 18.102
libavcodec     58. 19.102 / 58. 19.102
libavformat    58. 14.100 / 58. 14.100
libavdevice    58.  4.100 / 58.  4.100
libavfilter     7. 23.100 /  7. 23.100
libswscale      5.  2.100 /  5.  2.100
libswresample   3.  2.100 /  3.  2.100
libpostproc    55.  2.100 / 55.  2.100

C:\0>ffs Part_3.mp4 -i un.srt -o s.srt
Traceback (most recent call last):
  File "C:\Python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python36\Scripts\ffs.exe\__main__.py", line 4, in <module>
  File "C:\Python36\lib\site-packages\ffsubsync\__init__.py", line 6, in <module>
    from rich.console import Console
  File "C:\Python36\lib\site-packages\rich\console.py", line 44, in <module>
    from . import errors, themes
  File "C:\Python36\lib\site-packages\rich\themes.py", line 1, in <module>
    from .default_styles import DEFAULT_STYLES
  File "C:\Python36\lib\site-packages\rich\default_styles.py", line 19, in <module>
    strike=False,
  File "C:\Python36\lib\site-packages\rich\style.py", line 150, in __init__
    self._color = None if color is None else _make_color(color)
  File "C:\Python36\lib\site-packages\rich\style.py", line 148, in _make_color
    return color if isinstance(color, Color) else Color.parse(color)
AttributeError: type object 'Color' has no attribute 'parse'

C:\0>
`

**Comment by smacke** [comment, marked as answer]
https://github.com/Textualize/rich/issues/215

Looks like rich no longer supports Python 3.6. On my end, my options are either to fix it in rich, to make ffsubsync depend on an older version of rich on Python 3.6 only, or to stop supporting Python 3.6. Honestly I find the latter the most tempting.

At some point I'll see about fixing this on 3.6, but it will be a while. In the mean time, you'll have to upgrade to Python >= 3.7 for ffsubsync to work.

**Comment by Kid-C** [comment]
I appreciate that information - thanks.

I tried building today with ActiveState for Python 3.8 and 3.9 and I was getting build errors for both.  3.9 build log attached.

Sorry for your troubles with this. No urgency from my side.

[Python_3.9___ffsubsync_build_error___2022-12-20.txt](https://github.com/smacke/ffsubsync/files/10273499/Python_3.9___ffsubsync_build_error___2022-12-20.txt)
